src/promptflow-evals/promptflow/evals/evaluate/_utils.py

```diff
@@ -199,3 +202,13 @@ def _apply_column_mapping(source_df: pd.DataFrame, mapping_config: dict, inplace
 
 def _has_aggregator(evaluator):
     return hasattr(evaluator, "__aggregate__")
+
+
+def set_event_loop_policy():
+    import asyncio
+    import platform
+
+    if platform.system().lower() == "windows":
+        # Reference: https://stackoverflow.com/questions/45600579/asyncio-event-loop-is-closed-when-getting-loop
+        # On Windows seems to be a problem with EventLoopPolicy, use this snippet to work around it
+        asyncio.set_event_loop_policy(asyncio.WindowsSelectorEventLoopPolicy())

```

Comment:
According to the reference link, after python 3.7, we could use `asyncio.run` instead of `loop.run_until_complete()` to resolve this issue.

I believe we always expect customer to use python 3.7+ (not sure the precise version). So, maybe we could skip this platform specific handling here. (And recommend customer use `asyncio.run` if they meet the same issue)

And the only usage of run_until_complete is in adversarial_simulator.py. Guess we could get rid of it also.
https://github.com/microsoft/promptflow/blob/696b1293de12ee7391599902f80f65395fe71fc8/src/promptflow-evals/promptflow/evals/synthetic/adversarial_simulator.py#L372

Reply:
Thanks for pointing this out. However, the code we are changing here has nothing to do with the adversarial simulator, so that's not causing the problem. The issue could be due to the other external async library the evaluator depends on. For safety, we will still need this code to avoid running into the event loop closed issue.

Regarding the adversarial simulator, I will follow up with my colleague who owns that component to refine it if possible.